packages/RangeFilter/src/logic.tsx

```diff
@@ -146,55 +162,32 @@ const RangeFilter = function(currentElement: any) {
     slider.setAttribute("max-value", maxValue);
   };
 
-  function onStart(event: any) {
-    event.preventDefault();
-    let eventTouch = event;
-
-    if (event.touches) {
-      [eventTouch] = event.touches;
-    }
-
-    this.x = this === touchLeft ? touchLeft.offsetLeft : touchRight.offsetLeft;
-
-    this.startX = eventTouch.pageX - this.x;
-    selectedTouch = this;
-
-    slider.addEventListener("mousemove", onMove);
-    slider.addEventListener("mouseup", onStop);
-    slider.addEventListener("touchmove", onMove);
-    slider.addEventListener("touchend", onStop);
-  }
-
   const onMove = (event: any) => {
-    let eventTouch = event;
+    let eventTouch = event.touches ? ([eventTouch] = event.touches) : event;
 
-    if (event.touches) {
-      [eventTouch] = event.touches;
-    }
-
-    this.x = eventTouch.pageX - this.startX;
+    xAxis = eventTouch.pageX - startX;
 
     if (selectedTouch === touchLeft) {
-      if (this.x > touchRight.offsetLeft - selectedTouch.offsetWidth + MARGIN) {
-        this.x = touchRight.offsetLeft - selectedTouch.offsetWidth + MARGIN;
-      } else if (this.x < 0) {
-        this.x = 0;
+      if (xAxis > touchRight.offsetLeft - selectedTouch.offsetWidth + MARGIN) {
+        xAxis = touchRight.offsetLeft - selectedTouch.offsetWidth + MARGIN;
+      } else if (xAxis < 0) {
+        xAxis = 0;
       }
 
-      selectedTouch.style.left = `${this.x}px`;
+      selectedTouch.style.left = `${xAxis}px`;
     } else if (selectedTouch === touchRight) {
-      if (this.x < touchLeft.offsetLeft + touchLeft.offsetWidth - MARGIN) {
-        this.x = touchLeft.offsetLeft + touchLeft.offsetWidth - MARGIN;
-      } else if (this.x > maxX) {
-        this.x = maxX;
+      if (xAxis < touchLeft.offsetLeft + touchLeft.offsetWidth - MARGIN) {
+        xAxis = touchLeft.offsetLeft + touchLeft.offsetWidth - MARGIN;
+      } else if (xAxis > maxX) {
+        xAxis = maxX;
       }
-      selectedTouch.style.left = `${this.x}px`;
+      selectedTouch.style.left = `${xAxis}px`;
```

Comment:
To many ifs, cannot we try to simplify that?

Reply:
of course